src/main/java/com/mineboundteam/minebound/inventory/ArmorForgeMenu.java

```diff
@@ -0,0 +1,161 @@
+        return RecipeBookType.CRAFTING;
+    }
+
+    public int getResultSlotIndex() {
+        return 0;
+    }
+
+    public int getSize() {
+        return SIZE;
+    }
+
+    @Override
+    public @NotNull ItemStack quickMoveStack(@NotNull Player player, int index) {
+        int vanillaSlotCount = 36;
+        int vanillaIndex = 0;
+        int teInventoryIndex = vanillaIndex + vanillaSlotCount;
+        Slot slot = slots.get(index);
+        if (!slot.hasItem()) return ItemStack.EMPTY;
+        ItemStack sourceStack = slot.getItem();
+        ItemStack copyOfSourceStack = sourceStack.copy();
+        if (index < vanillaIndex + vanillaSlotCount) {
+            if (!moveItemStackTo(sourceStack, teInventoryIndex, teInventoryIndex + SIZE, false)) {
+                return ItemStack.EMPTY;
+            }
+        } else if (index < teInventoryIndex + SIZE) {
+            if (!moveItemStackTo(sourceStack, vanillaIndex, vanillaIndex + vanillaSlotCount, false)) {
+                return ItemStack.EMPTY;
+            }
+        } else {
+            return ItemStack.EMPTY;
+        }
+        if (sourceStack.getCount() == 0) {
+            slot.set(ItemStack.EMPTY);
+        } else {
+            slot.setChanged();
+        }
+        slot.onTake(player, sourceStack);
+        return copyOfSourceStack;
+    }
+
+    public boolean recipeMatches(Recipe<? super CraftingContainer> recipe) {
+        return recipe.matches(this.craftingContainer, this.player.level);
+    }
+
+    public boolean shouldMoveToInventory(int slotIndex) {
+        return slotIndex != this.getResultSlotIndex();
+    }
+
+    @Override
+    public void slotsChanged(@NotNull Container container) {
+        this.containerLevelAccess.execute((level, blockPos) -> {
+            slotChangedCraftingGrid(this, level, this.player, this.craftingContainer, this.resultContainer);
+        });
+    }
+
+    protected static void slotChangedCraftingGrid(ArmorForgeMenu armorForgeMenu, Level level, Player player, CraftingContainer craftingContainer, ResultContainer resultContainer) {
+        if (level.isClientSide) return;
+
+        ServerPlayer serverplayer = (ServerPlayer) player;
+        if (level.getServer() == null) return;
+
+        ItemStack itemStack = ItemStack.EMPTY;
+        Optional<ArmorForgeRecipe> optional = level.getServer().getRecipeManager().getRecipeFor(RecipeRegistry.ARMOR_FORGE_RECIPE.get(), craftingContainer, level);
+        if (optional.isPresent()) {
+            ArmorForgeRecipe armorForgeRecipe = optional.get();
+            if (resultContainer.setRecipeUsed(level, serverplayer, armorForgeRecipe)) {
+                itemStack = armorForgeRecipe.getResultItem();
+            }
+        }
+        resultContainer.setItem(0, itemStack);
+    }
+
+    public boolean stillValid(@NotNull Player player) {
+        return true;
+    }
+}
+
+class InputArmorSlot extends Slot {
+    public InputArmorSlot(Container container, int slotIndex, int x, int y) {
```

Comment:
Can you add an override here to make sure no items that aren't of type MyrialArmorItem can't be put into this slot? I'm pretty sure slot has overrides like that